---
title: "The impact of groundwater temperature on the composition of groundwater microbial communities in an urban aquifer" 
author:  "Angela Cukusic"
output:
  html_document:
    toc: true               # Table of contents
    toc_float: true         # Keep the table of contents floating on the page
    code_download: true     # Enable code download button
    code_copy: true         # Enable code copy button
    number_sections: true   # Numbering the sections
    theme: united           # Choose a Bootstrap theme (check available themes)
    highlight: tango        # Syntax highlighting style 

---

```{r setup, include=FALSE}
# For width of code chunks and scroll bar 
options(width=250)

knitr::opts_chunk$set(eval = TRUE, 
                      echo = TRUE, 
                      cache = TRUE,
                      include = TRUE,
                      warning = FALSE,
                      collapse = FALSE,
                      message = FALSE,
                      engine = "R", # Chunks will always have R code, unless noted
                      error = TRUE)

```


# load packages
```{r load-libraries, message = FALSE, warning = FALSE} 
library(readxl)
library(tidyverse)
library(stringr)

library(decontam)     # removing contaminants
library(SRS)
library(vegan)
library(patchwork)    # combining plots in a single figure
library(ggpmisc)
library(ggpubr)       # cleaner plotting 
library(MASS)
library(interactions)
library(microViz)
library(microbiome)
library(ggstatsplot)
library(mapview)      # coordianate plotting
library(sf)           # coordianate plotting
library(geosphere)    # coordianate plotting

# Set the ggplot theme 
theme_set(theme_linedraw())
```


# PART I: data clean up and preparation


## metadata  
```{r load-data, eval = TRUE, warning = FALSE}


# each campaign has its dataset, and each of them has 4 sheets
## combine them

#SPRING XLSX
spring_chem <- read_excel("C:/Users/Angela Cukusic/Desktop/DS_analysis/data/HBTC_mt_spring.xlsx")
spring_sampl <- read_excel("C:/Users/Angela Cukusic/Desktop/DS_analysis/data/HBTC_mt_spring.xlsx", 
                           sheet = "Sampling Info")
spring_ATP <-  read_excel("C:/Users/Angela Cukusic/Desktop/DS_analysis/data/HBTC_mt_spring.xlsx", 
                          sheet = "ATP")
spring_TCC <-  read_excel("C:/Users/Angela Cukusic/Desktop/DS_analysis/data/HBTC_mt_spring.xlsx", 
                          sheet = "TCC") 

#FALL XLSX
fall_chem <- read_excel("C:/Users/Angela Cukusic//Desktop/DS_analysis/data/HBTC_mt_autumn.xlsx", 
                        sheet = "Chemistry")
fall_sampl <- read_excel("C:/Users/Angela Cukusic/Desktop/DS_analysis/data/HBTC_mt_autumn.xlsx", 
                         sheet = "Sampling Info")
fall_ATP <-   read_excel("C:/Users/Angela Cukusic/Desktop/DS_analysis/data/HBTC_mt_autumn.xlsx",  
                         sheet = "ATP")
fall_TCC <-   read_excel("C:/Users/Angela Cukusic/Desktop/DS_analysis/data/HBTC_mt_autumn.xlsx", 
                         sheet = "TCC")

# add this column as a unique identifier, to differentiate spring and fall wells
# add to each sheet in order to combine them in one dataset
spring_chem$sample_season <- paste(spring_chem$`Sample ID`, "spring", sep = "_")
spring_sampl$sample_season <- paste(spring_sampl$`Sample ID`, "spring", sep = "_")
spring_ATP$sample_season <- paste(spring_ATP$`Sample ID`,  "spring", sep = "_")
spring_TCC$sample_season <- paste(spring_TCC$`Sample ID`,"spring", sep = "_")

fall_chem$sample_season <- paste(fall_chem$`Sample ID`,  "fall", sep = "_")
fall_sampl$sample_season <- paste(fall_sampl$`Sample ID`,  "fall", sep = "_")
fall_ATP$sample_season <- paste(fall_ATP$`Sample ID`, "fall", sep = "_")
fall_TCC$sample_season <- paste(fall_TCC$`Sample ID`, "fall", sep = "_")



### sheets

# 1. chemical info
spring_chem <- spring_chem %>% dplyr::select("Sample ID", "sample_season", "DIC (mg/L)", 
                                             "DOC (mg/L)", "PO4 (µg/L)",  "SO4 (mg/L)" , 
                                             "Cl (mg/L)" , "Na (mg/L)", "K (mg/L)", 
                                             "Ca (mg/L)" , "Mg (mg/L)"  )
fall_chem <- fall_chem %>%  dplyr::select("Sample ID", "sample_season", "DIC (mg/L)", 
                                          "DOC (mg/L)", "PO4 (µg/L)", "SO4 (mg/L)", 
                                          "Cl (mg/L)", "Na (mg/L)", "K (mg/L)", 
                                          "Ca (mg/L)", "Mg (mg/L)" )

# 2. sampling info                                      
spring_sampl <- spring_sampl %>%  dplyr::select("sample_season", "Land use", "Date", "Rohr D dm",  
                                                "GW Table [m]", "Well depth [m]", 
                                                "Watercolumn [m]", "Extraction depth [m]", 
                                                "T_3 (C)", "Lf_3 (uS)", "O2_3 (mg)",  "pH_3")
fall_sampl <- fall_sampl %>%  dplyr::select("sample_season", "Land use" , "Date","Rohr D dm", 
                                            "GW Table [m]", "Well depth [m]", 
                                            "Watercolumn [m]", "Extraction depth [m]", 
                                            "T_3 (C)", "Lf_3 (uS)", "O2_3 (mg)",  "pH_3")


# 3. ATP info
spring_ATP <- spring_ATP %>%  rename("tot ATP (pM)" = "tot ATP (pM)...3" , 
                                     "ext ATP" = "ext ATP...5", 
                                     "int ATP" = "int ATP...8") %>% 
  dplyr::select("sample_season", "tot ATP (pM)", 
                "ext ATP", "int ATP")
fall_ATP <- fall_ATP %>%   dplyr::select("sample_season", "tot ATP (pM)", "ext ATP", 
                                         "corr. int ATP" ) %>%
  rename("int ATP" = "corr. int ATP")

# 4. TCC info
spring_TCC <- spring_TCC %>%  dplyr::select("sample_season", "TCC (MW, Cells/ml)") %>% 
  rename( "TCC" = "TCC (MW, Cells/ml)")
fall_TCC <- fall_TCC %>%  dplyr::select("sample_season",  "TCC (MW, Cells/ml)") %>% 
  rename("TCC" = "TCC (MW, Cells/ml)")

## join chemical, sampling info, ATP and TCC 
df1<- inner_join(spring_chem, spring_sampl, by="sample_season") %>% 
      left_join(.,spring_ATP, by="sample_season")  %>% 
      left_join(.,spring_TCC, by="sample_season") 

df2 <- inner_join(fall_chem, fall_sampl, by="sample_season") %>% 
  left_join(.,fall_ATP, by="sample_season")  %>% 
  left_join(.,fall_TCC, by="sample_season")  



#main sheet = data sheet with sample names and their JMF.id
main_sheet <- read.csv("C:/Users/Angela Cukusic/Desktop/DS_analysis/data/main_sheet.csv")
main_sheet$sample_season <- paste(main_sheet$Sample.ID, 
                                  main_sheet$Season, sep = "_")


# join sample info with its JMF.id
df1.2<- inner_join(main_sheet, df1, 
                   by=c('sample_season'='sample_season')) #185/187
df2.2 <- inner_join(main_sheet, df2, 
                    by=c('sample_season'='sample_season')) #183/163 (the 20 are reps?)
df3<- full_join(df1.2, df2.2, by = c("JMF.ID", "Sample ID", "sample_season", "Date", 
                                     "DIC (mg/L)",  "DOC (mg/L)", 
                                     "PO4 (µg/L)", "SO4 (mg/L)", "Cl (mg/L)",
                                     "Na (mg/L)", "K (mg/L)", "Ca (mg/L)", "Mg (mg/L)",
                                     "Land use", "Rohr D dm", "GW Table [m]",
                                     "Well depth [m]", "Watercolumn [m]", 
                                     "Extraction depth [m]", "T_3 (C)", "Lf_3 (uS)", 
                                     "O2_3 (mg)", "pH_3", "tot ATP (pM)", "ext ATP", 
                                     "int ATP", "TCC") )
                                     
metadata <- df3 
# order it from JMF-001 to JMF-410
metadata<-metadata[order(metadata$JMF.ID),] 
rownames(metadata) <- metadata$JMF.ID


metadata <- metadata %>% rename("Sample_ID" = "Sample ID",
                                "DIC" = "DIC (mg/L)", 
                                "DOC" = "DOC (mg/L)",
                                "PO4" = "PO4 (µg/L)",
                                "SO4" ="SO4 (mg/L)", 
                                "Cl"=  "Cl (mg/L)",  
                                "Na" = "Na (mg/L)", 
                                "K" ="K (mg/L)", 
                                "Ca" = "Ca (mg/L)", 
                                "Mg" = "Mg (mg/L)",  
                                "Land_use" = "Land use", 
                                "Ex_depth" =  "Extraction depth [m]", 
                                "Depth" = "Well depth [m]",
                                "Temp" =  "T_3 (C)", 
                                "Lf"=  "Lf_3 (uS)", 
                                "O2"= "O2_3 (mg)", 
                                "pH"="pH_3", 
                                "totATP" = "tot ATP (pM)", 
                                "extATP"= "ext ATP", 
                                "intATP" = "int ATP" ) %>% 
  dplyr::select("JMF.ID",  "sample_season", "Sample_ID", "Date",  "Land_use", "DIC", "DOC", "PO4", "SO4", 
                "Cl" , "Na", "K",  "Ca", "Mg", "Depth", "Temp", "Lf",  
                "O2", "pH", "intATP", "TCC")




rm(df1, df2, fall_ATP, fall_chem, fall_sampl, fall_TCC, spring_ATP, 
   spring_chem, spring_sampl, spring_TCC)
rm( df1.2, df2.2)

```



## categorical data 

```{r}
# 1. heavy metal data in fall dataset
add_data1 <- read_excel("C:/Users/Angela Cukusic/Desktop/DS_analysis/data/fall_data.xlsx")
add_data1 <- add_data1 %>% dplyr::select("Sample ID", "Probennr","UHI",  
                     "HBTC_Kam_all_ug Fe/l","Name_HBC",     
                     "HBTC_Kam_all_NO3 (Nitrat)", 
                     "HBTC_Kam_all_NH4 (Ammonium)",  
                     "DOM_bix") %>% 
              rename("Sample_ID" = "Sample ID",   
                     "Fe" = "HBTC_Kam_all_ug Fe/l",     
                     "NO3" = "HBTC_Kam_all_NO3 (Nitrat)",   
                     "NH3" = "HBTC_Kam_all_NH4 (Ammonium)")
# add season-specific identifier
add_data1$sample_season <- paste(add_data1$Probennr,  "fall", sep = "_")

# add to metadata
metadata <- metadata %>% left_join(., add_data1, by = "sample_season")


# 2. second dataset has aquifer categorization, land use, type, OF_beb, coordinates
add_data2 <- read.csv("C:/Users/Angela Cukusic/Desktop/DS_analysis/data/spring_data.csv", row.names=NULL, sep=";")
## but it does not have the same well identifier as metadata 
## first add column with "Sample ID" from add_data1

add_data2 <- add_data2 %>% dplyr::select("Probennr", "X", "Y",  # coordinates
                     "Land.use", "Geol_beb", "Type", "OF_beb")
# look if they are all a factor
#OF_beb is not
add_data2$OF_beb <- as.factor(add_data2$OF_beb) 

# not adding season-specific identifier because these categorizations are per well
# add to metadata
metadata <- metadata %>% left_join(., add_data2, by = c("Sample_ID.x" = "Probennr"))


# we are left with data on surface water as well
# best bet to remove is to filter out samples with not aquifer type category
metadata <- metadata %>% dplyr::filter(Geol_beb != "" ) %>% 
                          dplyr::filter(Geol_beb != "OF" ) 
                          #for some reason doesnt work with %in% c("", "OF") 


rm( add_data2)

```


# PART II: remove contaminants, average replicates, rarify

## asv table, taxonomy, blanks
```{r}
# 2. load asv_table ####
asv_table <- read.delim("C:/Users/Angela Cukusic/Desktop/DS_analysis/data/DADA2_counts_as_matrix.tsv", row.names=1)
names(asv_table)<- sub("^JMF.2207.07.0","JMF-2207-07-0", names(asv_table))
names(asv_table) <- sub("A", "", names(asv_table))


# load taxonomy ####
asv_taxonomy <-
  read.csv("C:/Users/Angela Cukusic/Desktop/DS_analysis/data/DADA2_ASVs.rRNA_SSU.SILVA_classified.csv")
asv_taxonomy <- asv_taxonomy %>% dplyr::select("name", "lca_tax_slv") %>%
  separate_wider_delim(lca_tax_slv, delim = ";", 
                       # too few can be c("error", "debug", "align_start") and
                       # means there is  less taxa names (NAs/not classified) 
                       too_few = "align_start", 
                       # too many can be c("error", "debug", "drop") and 
                       # means there is more taxa names that levels (ssp.)
                       too_many = "drop",
                       names = c("Kingdom", "Phylum", "Class", "Order", 
                                 "Family", "Genus", "Species")) %>% 
          column_to_rownames("name")
# with the separate_wider_column the first NA in line end up just empty
## so make that an NA as well:
asv_taxonomy[asv_taxonomy == ''] <- NA



# load the blank_sheet which has info on blank batches
blanks_df <- read.delim("C:/Users/Angela Cukusic/Desktop/DS_analysis/data/blanks.csv",
                        header=TRUE, sep = ",")


```


##decontamination
```{r}
#| warning: false
#| output: false

# i have one blank which was not sequenced so i removed the whole batch :(
## 12 samples - not optimal!
blanks_all <-  blanks_df %>% filter(batch_name !="Blank_26")
# remove the 12 samples from the asv_table as well
asv_all <- asv_table[ , which(colnames(asv_table) %in% blanks_all$JMF.sample.ID)]  



# see if all samples match from blank_df to asv_table - are all the JMFs 
## from the blank table also samples from the asv_table
blanks_all[which(!  blanks_all$JMF.sample.ID  %in% colnames(asv_all)), ]
# they dont match (JMF-2207-07-0282 was sequenced but not measured for env. data)
## remove it
blanks_all <- blanks_all %>% filter(JMF.sample.ID != "JMF-2207-07-0282") 


# (before using decontam it is important that rows are ordered in 
## the same order as columns of asv_table!)
## in our case they are because the JMF.ID goes up by number


# STEP 1: make vector for blanks 
vector_for_decontam <- print(blanks_all$sample_blank=="blank") 
# STEP 2: identify
contam_df <- isContaminant(t(asv_all), 
                           neg=vector_for_decontam, 
                           batch = blanks_all$batch_name)


```


```{r}
#| results: hold

# identified 5763  as contaminants
table(contam_df$contaminant) # to see how many are contaminants
# name the contaminants
contam_asvs <- row.names(contam_df[contam_df$contaminant == TRUE, ])
contam_asvs <- contam_asvs %>% as.data.frame() 
# find the taxonomy of the contaminats
cont_identified <- asv_taxonomy %>% rownames_to_column(".") %>% 
  left_join (contam_asvs, ., by=".") 
table(cont_identified$Class) %>% as.data.frame() %>%  arrange(desc(Freq)) %>% head()
# the most frequent contamints are Nanoarchaea


# Step 3: remove
# remove asvs which are contaminants from the asv_table
asv_all_no_cont <- asv_all[!row.names(asv_all) %in% contam_asvs, ]
# removes blanks samples (JMFs) from the asv_table
asv_all_no_cont <- asv_all_no_cont[ , !vector_for_decontam]
# removes contaminants from taxonomy table
new_taxonomy <- asv_taxonomy[!row.names(asv_taxonomy) %in% contam_asvs, ]


# Step 4: clean the rest 
# be sure that metadata has the same samples as the final asv table
rownames(metadata) <- metadata$JMF.ID
new_metadata <- metadata[which(rownames(metadata) %in% colnames(asv_all_no_cont) ),]
#and that all the samples in asv_table are in the metadata as well
asv_all_no_cont <- asv_all_no_cont[, which( colnames(asv_all_no_cont)  %in%  rownames(new_metadata))]


rm(blanks_all, blanks_df, contam_asvs, asv_taxonomy, asv_table)

```

## replicate averaging

```{r}
#| output: false
#| warning: false 

#it should be easy to do this with phyloseq BUT it is not!
# ps_all <- merge_samples(ps_all, "sample_season", fun = mean)
# it eather a: changes all the factors to numeric and it is not so easy to revert
# the categories (i.e. each aquifer is now assigned a number)
# or b) it is straigh up just NAs



# this way is taxing and long, but for now I do not have a better alternative:

## a) metadata without replicates ------------------------
# remove the duplicate env data for replicates of the same sample:
# first see which wells have replicates 
mt_no_rep <- main_sheet %>%
  # main_sheet has JMF to sample name identifier
  filter(!grepl('Rep2|Rep3|Rep4', Sample.ID.with.replicates))
# remove all the names of replicates only keeping Rep1

# keep in metadata the well measurement only for the Rep1
new_metadata <- new_metadata[which(new_metadata$JMF.ID %in% mt_no_rep$JMF.ID),] 


## b) asv_table replicates ---------------------------
# now average Replicates by mean, and keep only the Rep1 JMFs to coincide with the metadata

# b.1
asv_ohne_rep <- asv_all_no_cont %>%
  rownames_to_column("ASV") %>%
  # pivot longer so you can summarize and add metadata (sample_season identifier)
  pivot_longer(cols= - ASV, names_to = "samples", values_to = "abundance") %>%
  left_join(., new_metadata, by=c("samples"="JMF.ID") ) %>%
  dplyr::select("ASV", "abundance", "sample_season") %>%
  # apply function (summary) for each individual ASV per sample_season
  # replicates of same well have the same sample_season 
  group_by(ASV, sample_season) %>%
  # find abund means of samples of the same well (same sample_season)
  summarise(abundance = mean(abundance)) %>% 
  ungroup()

# b.2
asv_all_no_rep <- main_sheet %>%  # keep just the Rep1
  filter(!grepl('Rep2|Rep3|Rep4', Sample.ID.with.replicates))%>%
  dplyr::select("sample_season","JMF.ID") %>%
  right_join(., asv_ohne_rep, by="sample_season") %>%
  dplyr::select(!"sample_season")

# order it from JMF-001 to JMF-410
asv_all_no_rep <- asv_all_no_rep[order(asv_all_no_rep$JMF.ID),] 
# easier to order it before transferring back to wide format than ordering 
## column names

# back to wide format
asv_table_final <- asv_all_no_rep %>%  
  pivot_wider(names_from = "JMF.ID", values_from = "abundance") %>%
  column_to_rownames(var="ASV")



# did any samples get lost 
new_metadata[which(!new_metadata$JMF.ID %in% colnames(asv_table_final)), ] 
asv_table_final <- asv_table_final[,which(colnames(asv_table_final) %in% rownames(new_metadata))] 

# remove to keep environment clearer
rm(mt_no_rep, main_sheet, asv_ohne_rep)

```


## sequence numbers 
```{r}
#| results: hold
read_data <- asv_table_final %>% 
colSums() %>% # how many reads per sample (find sums per each column)
as.data.frame() %>% 
rownames_to_column("JMF.sample.ID") %>% 
rename("reads" = ".") 

summary(read_data)
sd(read_data$reads)

# mean: 7374   +- 4127.732
```


# PART III:

```{r}
# color palettes
pal3 = c("#253494", "#225ea8", "#1d91c0", "#7fcdbb" , "#fdbb84", "#d7301f", "#7f0000", "gray")

c25 <- c("dodgerblue2", "#E31A1C", "green4" ,  "#6A3D9A",     "skyblue2", 
         "blue1", "gold1", "brown",  "#FB9A99", "steelblue4" ,  "palegreen2" , 
         "#CAB2D6", "#FDBF6F", "khaki2",  "maroon",  "gray70",  "orchid1",  
         "deeppink1",  "darkorange4",  "green1",  "yellow4"  ,    "yellow3",  
         "beige", "darkturquoise", "#FF7F00" )

```


## temp categorization 
```{r}

# season variable
new_metadata  <- new_metadata %>% 
  mutate(season = case_when( str_detect(sample_season, "_fall") ~ "fall",
                             str_detect(sample_season,"_spring")~ "spring") ) 


# temperature categories
new_metadata$Cat3 <- cut(new_metadata$Temp, 
                         breaks=c(0, 10, 12, 14, 16, 18, 20, 30), 
                         labels=c('<10', '10-12', '12-14', '14-16', 
                                  '16-18', '18-20', '20<'))
# find reference samples (the one in Marchfeld)
reference <- new_metadata %>% 
  filter(sample_season %in% c("22-246_fall", "EM10_spring",  "EM62_spring", "22-52_fall", 
                              "22-52_spring", "22.7/2_spring", "22.7/2_fall"))

# distinguish reference samples from the 10-12C
new_metadata <- new_metadata %>%
  mutate( reference = case_when(sample_season %in% reference$sample_season ~ "referential", 
                                TRUE ~ "other")) 
new_metadata$ref_cat <- paste(new_metadata$Cat3, new_metadata$reference, sep="_")
new_metadata <- new_metadata %>% 
  mutate( category = case_when(ref_cat == "<10_other" ~ "<10",
                               ref_cat == "10-12_other" ~ "10-12",
                               ref_cat == "12-14_other" ~ "12-14",
                               ref_cat == "14-16_other" ~ "14-16",
                               ref_cat == "16-18_other" ~ "16-18",
                               ref_cat == "18-20_other" ~ "18-20",
                               ref_cat == "20<_other" ~ "20<",
                               TRUE ~ "ref_cat")) %>% 
  dplyr::select(!c("Cat3" , "reference" ,    "ref_cat"))
#at this point 188 samples for just the DS, and 335 for all the wells ?
```



## rarefaction 
```{r}

# option a) vegan
asv_table_final <- asv_table_final[, colSums(asv_table_final) >= 2000] 
# remove those that have less than 2000
srs_asv_all <- rrarefy(t(asv_table_final), sample = 2000) %>% t() %>% as.data.frame()
new_metadata <- new_metadata[which(new_metadata$JMF.ID %in% colnames(srs_asv_all)  ),]

# option b) srs
#srs_asv_all <- SRS(asv_table_final, 2000) #rarifying to 2000 reads
#rownames(srs_asv_all) <- rownames(asv_table_final)
#new_metadata <- new_metadata[which(new_metadata$JMF.ID %in% colnames(srs_asv_all)  ),]


#final number of samples is 152!
```




## phyloseq making 
```{r}

# how many taxa are now no longer in the asv_table after subsetting for DS ??
#as.data.frame(rowSums(srs_asv_all)) %>% filter(rowSums(srs_asv_all)  == "0") %>% nrow()
# 4765
ps_all <- phyloseq(otu_table(as.matrix(srs_asv_all), taxa_are_rows=TRUE), 
                   tax_table(as.matrix(new_taxonomy) ) , 
                   sample_data(as.data.frame(new_metadata)) )

ps_all <- prune_taxa(taxa_sums(ps_all) > 0, ps_all)

```

## alpha diversity

### temperature distribution
```{r}
#| results: hold

# or with vegan 
shannondiv <- vegan::diversity(t(srs_asv_all))
ens <- exp(shannondiv)

p1_en <- cbind(as.data.frame(ens), as.factor(new_metadata$category)) %>% 
  as.data.frame() %>%
  rename("div"= "ens","temp"="as.factor(new_metadata$category)") %>%
  filter(div > 0) %>% 
  ggplot( aes(x = temp, y = div, fill = temp, color=temp)) +
  #stat_boxplot(geom ='errorbar') + 
  geom_boxplot(alpha=0.5) +
  scale_fill_manual(values = pal3)+
  scale_color_manual(values = pal3)+
  theme_bw()+
  theme(legend.position = "none",
        axis.title=element_text(size=14,face="bold"),
        axis.text.y = element_text(size=10,face="bold"),
        axis.text.x = element_text(angle = 90)) +
  labs(x = "Temperature category",
       y = "Shannon diversity index") +
  scale_x_discrete(limits = c( "ref_cat", "<10", "10-12", "12-14", "14-16", 
                               "16-18",  "18-20",  "20<" ))  

p1_en

```
testing
```{r}
#| results: hold

# since there is only 1 sample in the 20< i must remove it before testing
data_for_alpha <- cbind( as.data.frame(ens), new_metadata$category) %>% 
  filter(new_metadata$category != "20<") %>% 
  rename("category" = "new_metadata$category")


# perform ANOVA?
## 1. assumption: Normal distribution: 
## Each population to be compared should be normally distributed (NOT THE ENTIRE DATASET).
shapiro.test (  data_for_alpha[which(data_for_alpha$category == "<10"), "ens"] )
# W = 0.89814, p-value = 0.3997
shapiro.test (  data_for_alpha[which(data_for_alpha$category == "10-12"), "ens"] )
# W = 0.77218, p-value = 0.04715 !!!
shapiro.test (  data_for_alpha[which(data_for_alpha$category == "12-14"), "ens"] )
# W = 0.97895, p-value = 0.4823
shapiro.test (  data_for_alpha[which(data_for_alpha$category == "14-16"), "ens"] )
# W = 0.9595, p-value = 0.05755
shapiro.test (  data_for_alpha[which(data_for_alpha$category == "16-18"), "ens"] )
# W = 0.93248, p-value = 0.1543
shapiro.test (  data_for_alpha[which(data_for_alpha$category == "18-20"), "ens"] )
# W = 0.58055, p-value = 0.0003393 !!!
shapiro.test (  data_for_alpha[which(data_for_alpha$category == "ref_cat"), "ens"] )
# W = 0.981, p-value = 0.9642

# would be better to do a non-parametric test

## 2. assumption: Homogeneity of variance: 
## Variance in the populations compared should be the same/similar. 
bartlett.test(ens ~ category, data= data_for_alpha)
# Bartlett's K-squared = 0.94125, df = 6, p-value = 0.9877
# assumption met

kruskal.test (ens ~ category, data= data_for_alpha) 
# Kruskal-Wallis chi-squared = 11.995, df = 6, p-value = 0.06208



```

```{r}

library(afex) ## to run ANOVA

alph <- cbind(as.data.frame(ens), as.factor(new_metadata$category)) %>% 
                as.data.frame() %>%
                rename("div"= "ens","temp"="as.factor(new_metadata$category)") 

ggbetweenstats( data  = alph, x = temp, y = div,
                type = "np",
                xlab = "Temperature category", 
                ylab = "Shannon diversity index",
                ggtheme = ggplot2::theme_classic(),
                title = "Comparison of microbial diversity across temperature increase",
                #boxplot.args = list(width = 0),
                #violin.args = list(width = 3, alpha = 4),
                  point.args = list(position = ggplot2::position_jitterdodge
                                    (dodge.width = 1), alpha = 0.4, 
                                    size = 3, stroke = 2, na.rm = TRUE),
                var.equal = TRUE) + 
  
  ggplot2::scale_x_discrete(limits = c( "ref_cat", "<10", "10-12", "12-14", "14-16", 
                               "16-18",  "18-20",  "20<" )) +
  ggplot2::scale_color_manual(values=pal3)
```


linear trend
```{r}
p1_en_li <- cbind(as.data.frame(ens), new_metadata$Temp) %>% 
  as.data.frame() %>%
  rename("div"= "ens","temp"="new_metadata$Temp") %>%
  ggplot(aes(temp,div)) + 
  geom_point(shape=21) + 
  geom_smooth(method = "lm", se= FALSE, color = "blue") + 
  stat_poly_eq(use_label(c("eq", "adj.R2", "p")), label.y = "bottom") + 
  theme_bw() + 
  labs(y = "Shannon diversity index", x ="Temperature")

p1_en_li
```

### spatial distribution
```{r}

# add mapview to change X and Y to standard crs
new_metadata2 <- new_metadata %>% 
  cbind(new_metadata, as.data.frame(ens)) %>% 
  dplyr::select("ens", "X", "Y") %>% drop_na() %>% 
  mutate( ens_cat = cut(ens, b=6) )  

pal <-  mapviewPalette("mapviewSpectralColors")

sf_data <- st_as_sf(new_metadata2, coords = c("X", "Y"),  crs = 31253)
mapview(sf_data,  map.types = "Stamen.Terrain", col.regions = pal(10),
        zcol= "ens_cat", zoom = 12, color="black", cex = 4, alpha = 0.9 ) 
```


### aquifer distributuion
```{r}
#|warning: false

# ANY AQUIFER THAT HAS MORE THAN 5 POINTS
cbind(new_metadata, ens ) %>% 
  dplyr::filter(! Geol_beb %in%  c("n.b.", "lokaler Aquifer",
                                   "Wiental", "Zubringer")) %>% 
                mutate(Geol_beb = case_when( grepl('DS', Geol_beb) ~ "Donauschotter", 
                                              TRUE ~ Geol_beb)) %>% 
  ggplot(aes(x=Temp, y=ens)) + 
  geom_point(shape=21, size = 2) + 
  geom_smooth(method = "lm", se= FALSE, color = "blue") + 
  facet_wrap(~ Geol_beb, scales = "free") + 
  #stat_poly_line() +
  stat_poly_eq(use_label(c("eq", "adj.R2", "p")), label.y = "bottom",
               size = 3.4) + 
  theme_bw() + 
  labs(x = "Temperature",
       y = "Effective number of species") 

```


### modelling 
```{r}
#| results: hold
 
data_for_reg <- cbind(new_metadata) %>% dplyr::select(where(is.numeric)) %>% 
  scale(center = T, scale = T) %>% cbind(., ens) %>% data.frame() %>% 
  dplyr::select(! c("DIC", "SO4", "Cl", "Na", "Ca", "Mg", "Lf",
                    "NO3", "NH3", "Fe", "X", "Y", "DOM_bix") )


nullModel = lm(ens ~ 1, data=data_for_reg) # model with the intercept only
fullModel = lm(ens ~ ., data=data_for_reg) # model with all  variables

#choose
step.model<-stepAIC(fullModel, # start with a model containing no variables
                    direction = 'both', # run forward selection
                    trace = 0) # dont show step-by-step process of model selection
#summary(step.model) # DOC, K, Depth
RsquareAdj(step.model) # 0.043-> 4.3 %
summary(lm(ens ~ DOC + K + Depth, data = data_for_reg)) 


plot(ens ~ DOC , data = data_for_reg) + abline(lm(ens ~ DOC , data = data_for_reg))
plot(ens ~  K , data = data_for_reg)+ abline(lm(ens ~ K , data = data_for_reg))
plot(ens ~  Depth, data = data_for_reg) + abline(lm(ens ~ Depth , data = data_for_reg))

```


## composition

### bar plot
```{r}
#| warning: false

# with microviz

#ps_all %>% tax_fix(unknowns = c("uncultured")) %>% 
#  comp_barplot("Class", n_taxa = 15, merge_other = TRUE, label = NULL) +
#  facet_wrap(vars(category), scales = "free", nrow = 2, ncol = 4) + 
#  coord_flip() +
#  scale_fill_manual(values=c25) + 
#  theme_minimal()




srs_asv_all %>%
  mutate(class = new_taxonomy$Class) %>%
  pivot_longer(cols = - class, values_to = "abundance", names_to = "sample") %>%
  left_join(.,new_metadata, by = c("sample" = "JMF.ID")) %>%  
  dplyr::select ("sample", "abundance", "class", "category") %>%
  group_by(category) %>%
  mutate (rel_abund = abundance / sum(abundance)) %>%
  ungroup() %>% 
  group_by(category, class) %>%
  summarise(rel_abund = sum(rel_abund)) %>%
  ungroup() %>%
  group_by(category, class) %>%
  mutate(class = case_when(rel_abund< 0.015~ "Diverse others", 
                           TRUE ~ class)) %>%
  ungroup() %>%
  group_by(category, class) %>%
  summarise(rel_abund = sum(rel_abund)) %>%
  ungroup() %>%
  #mutate(class = case_when( class == "" ~ "Unclassified", TRUE ~ class)) %>% 
  replace_na(list(class ='Unclassified' )) %>% 
  ggplot(aes(x = category, y = rel_abund, fill = class)) + 
  geom_bar(stat = "identity") +
  scale_fill_manual(values=c25) +
  theme_classic() +
  theme(panel.grid = element_blank(), legend.position="right") + 
  labs(x = "",
       y = "Relative abundance", 
       fill = "Class") +
  guides(fill=guide_legend(ncol =1)) +
  scale_x_discrete(limits = c( "ref_cat", "<10", "10-12", "12-14", "14-16", "16-18",  "18-20",  "20<" )) 

```

### bubble plot
```{r}
#| warning: false


data <- srs_asv_all %>%
  mutate(class = new_taxonomy$Class) %>%
  pivot_longer(cols = - class, values_to = "abundance", names_to = "sample") %>%
  left_join(.,new_metadata, by = c("sample" = "JMF.ID")) %>%  
  dplyr::select ("sample", "abundance", "class", "category") %>%
  group_by(category) %>%
  mutate (rel_abund = abundance / sum(abundance)) %>%
  ungroup() %>% 
  group_by(category, class) %>%
  summarise(rel_abund = sum(rel_abund)) %>%
  ungroup() %>%
  group_by(category, class) %>%
  mutate(class = case_when(rel_abund< 0.015~ "Diverse others", 
                           TRUE ~ class)) %>%
  ungroup() %>%
  group_by(category, class) %>%
  summarise(rel_abund = sum(rel_abund)) %>%
  ungroup() %>%
  #mutate(class = case_when( class == "" ~ "Unclassified", TRUE ~ class))
 replace_na(list(class ='Unclassified' ))

data %>% as.data.frame() %>% 
  ggplot(aes(x = as.factor(category), y = as.factor(class) )) + 
  geom_point(aes(size = rel_abund, fill=class ), alpha = 0.75, shape = 21) + 
  scale_size_continuous(limits = c(0.001, 0.5), 
                        range = c(1,17), 
                        #breaks = c(0.1, 0.2, 0.5)
  ) + 
  labs( x= "", y = "", size = "Relative Abundance", fill = "")  + 
  theme(legend.key=element_blank(), 
        axis.text.x = element_text(colour = "black", size = 12, face = "bold", 
                                   angle = 90, vjust = 0.3, hjust = 1),
        axis.text.y = element_text(colour = "black", face = "bold", size = 11), 
        legend.text = element_text(size = 10, face ="bold", colour ="black"), 
        legend.title = element_text(size = 12, face = "bold"), 
        panel.background = element_blank(), panel.border = 
          element_rect(colour = "black", fill = NA, size = 1.2), 
        legend.position = "right") +  
  scale_fill_manual(values = c25, guide = FALSE) +   
  scale_y_discrete(limits = rev(levels(data$class))) +
  scale_x_discrete(limits = c( "ref_cat", "<10", "10-12", "12-14", "14-16", 
                               "16-18",  "18-20",  "20<" )) 

```


```{r}
#| warning: false

## mean + sd ####
ps_for_sd <- tax_glom(ps_all, "Class") %>% 
  transform_sample_counts(., function(x) x/sum(x))

tb <- psmelt(ps_for_sd) %>%
  as_tibble

tb0 <- tb %>%
  group_by(Class) %>%
  dplyr::summarize( Mean = mean(Abundance), SD = sd(Abundance) ) %>%
  ungroup() %>% 
  mutate( Mean_per = .$Mean *100, sd_per = .$SD *100 )  %>% 
  dplyr::filter( Mean_per > 1 ) %>% 
  left_join(., 
            distinct( new_taxonomy[, c("Kingdom", "Phylum", "Class")], Class , 
                      .keep_all = TRUE ) ,
            by="Class") 

head(tb0[order(tb0$Mean_per), c("Class", "Mean_per", "sd_per", "Phylum")])
head(tb0[, c("Class", "Kingdom", "Phylum")] )

```


## beta diversity


### temperature distribution

```{r}
#| warning: false
#| output: false

# tax clean
new_taxonomy %>% rownames_to_column("asv") -> tax_tab


# aggregate to class
srs_asv_all %>% 
  rownames_to_column("asv") %>% 
  pivot_longer(cols = -asv, names_to = "samples", values_to = "abundances") %>% 
   left_join(., tax_tab, by = "asv") %>% 
  dplyr::select("samples", "Class", "abundances") %>% 
  group_by(samples,  Class) %>% 
  summarise(sum(abundances)) %>% 
  ungroup() %>% 
  dplyr::filter(Class != "") %>% 
  pivot_wider(names_from = "samples", values_from = "sum(abundances)") %>% 
  column_to_rownames("Class") -> asv_aggregated



# do an ordination
nmds <- metaMDS(t(asv_aggregated), k = 2, autotransform = FALSE)

```

```{r}
#| warning: false

# plot it 
nmds_plot <- as.data.frame(nmds$points) %>% 
  cbind(., as.factor(new_metadata$category)) %>% 
  dplyr::rename("cat" = "as.factor(new_metadata$category)") %>% 
  ggplot(aes(x=MDS1,y=MDS2) ) +
  geom_point(aes( color= cat), alpha = 0.2, size = 3) +
  geom_point(aes( color= cat), shape = 21, size = 3) +  
  scale_color_manual(values=pal3) +
  theme_bw() + 
  annotate(geom="text",x=1.4, y=1, label="stress =", color="black") +   
  annotate(geom="text", x=1.7, y=1, label=round(nmds$stress,4), color="black") +
  stat_ellipse(aes(colour = cat), linewidth = 0.3) + 
  theme( #aspect.ratio = 1,
    #legend.position = c( .98,  .02),
    #legend.justification = c("right", "bottom"),
    legend.background = element_blank(),
    legend.box.background = element_rect(colour = "gray")) +
  labs(color = "Temperature \ncategory") +
  geom_vline(xintercept = c(0), color = "grey70", linetype = 2) +
   geom_hline(yintercept = c(0), color = "grey70", linetype = 2) 

nmds_plot
  
```


test ordination
```{r}
#| results: hold
# first test assumption: equal dispersion
disp <- betadisper( 
  vegdist(t(asv_aggregated), method = "bray"), 
  new_metadata$category ) 

#p>0.05 is what we need - homogenous variances
anova(disp) #no sign. differences in dispersion
#TukeyHSD(disp) # if there was sign, i would check with this which one are they

boxplot(disp, col=pal3) 


# now test actual ordination
permanova <- adonis2(t(asv_aggregated) ~ new_metadata$category, 
                    method = "bray" , permutations = 999)
permanova # 0.452



# if it was significant i would test which groups are significantly different
#pair_res <- pairwiseAdonis::pairwise.adonis2(t(asv_aggregated) ~ category, data = new_metadata)


```


### spatial distribution
```{r}
#| warning: false
gg <- new_metadata %>% st_as_sf(., coords = c("X", "Y"), crs = 31253) %>% 
      arrange(JMF.ID) %>%
      st_set_crs(31253) %>% 
      st_transform(4326) %>% 
      st_coordinates() %>%
      as.data.frame() %>%
      distm(., fun = distHaversine) %>%
      as.dist(.) %>% as.vector()

aa <- srs_asv_all %>% t() %>%
      vegdist(., method = "bray") %>%
      as.vector()


tt <-  new_metadata %>%
  dplyr::select("Temp") %>%
  dist(., method = "euclidean") %>%
  as.vector()


matr <- data.frame(gg,tt, aa)

ggplot(matr, aes(y = gg, x = aa)) + 
  geom_point(size = 3, alpha = 0.3, color="black", 
             shape = 21, fill= "blue",size =5 ) +
  geom_smooth(method = "lm", colour = "black", alpha = 0.2) +
  stat_cor(method = "spearman", color="black", label.y.npc="top", 
           label.x.npc = "left",size=4)+
  theme_bw() + 
  labs(x = "Community dissimilarity", y = "Physical Separation (km)") + 
  theme( axis.text.x = element_text(face = "bold",colour = "black", size = 12), 
         axis.text.y = element_text(face = "bold", size = 11, colour = "black"), 
         axis.title= element_text(face = "bold", size = 14, colour = "black"), 
         panel.background = element_blank(),
         legend.position = "none",
         panel.border = element_rect(fill = NA, colour = "black")) +
  scale_y_continuous(breaks = seq(0, 30000, by = 5000),
                     labels = function(x) x / 1000) 



```

### aquifer distribution
```{r}
#| results: hold
cbind(new_metadata, as.data.frame(nmds$points) ) %>% 
dplyr::filter(! Geol_beb %in%  c("n.b.", "lokaler Aquifer",
                                   "Wiental", "Zubringer")) %>% 
                mutate(Geol_beb = case_when( grepl('DS', Geol_beb) ~ "Donauschotter", 
                                              TRUE ~ Geol_beb)) -> data_aq


# first test assumption: equal dispersion
disp_aq <- betadisper( 
  vegdist(t(  asv_aggregated[,which(colnames(asv_aggregated) %in% rownames(data_aq))] ), 
          method = "bray"), 
          data_aq$Geol_beb ) 

#p>0.05 is what we need - homogenous variances
anova(disp_aq) #no sign. differences in dispersion
#TukeyHSD(disp) # if there was sign, i would check with this which one are they

boxplot(disp_aq) 


# now test actual ordination
permanova_aq <- adonis2(t( asv_aggregated[,which(colnames(asv_aggregated) %in% rownames(data_aq))] ) ~ 
                        data_aq$Geol_beb, 
                         method = "bray" , 
                        permutations = 999)
permanova # 0.477


```



## constrained ordination

first make a PCA of env variables to use them as another env variable
```{r}

```


```{r}
#| results: hold
# env clean

new_metadata %>% dplyr::select(where(is.numeric)) %>% 
                  dplyr::select(!c("Fe", "NO3", "NH3", "DOM_bix", "X", "Y")) %>% 
                  scale() %>% as.data.frame()  -> env.dat



# aggregate to class
sp.dat <- asv_aggregated # already done for nmds


# asv clean
sp.dat [, which  ( colnames(sp.dat) %in% rownames(env.dat)) ] -> sp.dat

# transform hellinger (or clr sometimes)
decostand(sp.dat, method = "hellinger") -> sp.dat
#clr(sp.dat) -> sp.dat


simpleRDA <- capscale(t(sp.dat) ~  ., data=env.dat ,
                 distance = "bray",
                 scaling = "sites")

# summary(simpleRDA)
#9.6% constrained but look at vif

env_vars_low_vif <- vif.cca(simpleRDA) %>% 
                    as.data.frame() %>% 
                    dplyr::rename("env" = ".") %>% 
                    dplyr::filter(env < 10) 

 env.dat[,which(colnames(env.dat) %in%  rownames(env_vars_low_vif) )] -> env.dat


# without them
simpleRDA <- capscale(t(sp.dat) ~  ., 
                      data=  env.dat ,
                 distance = "bray",
                 scaling = "sites")

#summary(simpleRDA) 
# now 5.6%

```

test it
```{r}


# Test of all canonical axes
anova.model <- anova.cca(simpleRDA, by='axis', step=1000)  # cap1 is sign, cap2 not

# test the whole model
anova.model2 <- anova.cca(simpleRDA, step=1000) #is sign
RsquareAdj(simpleRDA)$adj.r.squared # and explains 1.7%


# test env paameters
anova.model3 <- anova(simpleRDA, step=1000, by = "term") #Temp + DOC sign, and K + O2 borderline


# new model
simpleRDA <- capscale(t(sp.dat) ~  Temp +  DOC + K + O2, data=env.dat ,
                 distance = "bray",
                 scaling = "sites")

#summary(simpleRDA) 
# now 1.4%

```

```{r}
# vectors
ccavectors <- as.matrix(scores(simpleRDA, display = "bp", scaling = "sites")*40) %>% 
  t() %>% as.data.frame() %>% 
  rename("K**" = "K",
         "Temp." = "Temp",
         "O2*" = "O2") %>%
  t() %>% as.data.frame()

# site coordinates
site_data <- scores(simpleRDA, display = "sites") %>% 
  as.data.frame() %>% 
  cbind(., new_metadata$category) %>% 
  rename("cat" = "new_metadata$category")

# plotting
plot_cca <- 
  site_data %>% 
  ggplot( aes(x = CAP1, y = CAP2)) +
  geom_point(aes( color= cat), alpha = 0.2, size = 3) +
  geom_point(aes( color= cat), shape = 21, size = 3) +  
  geom_segment(data = ccavectors, aes(x = 0, y = 0, xend = CAP1, yend = CAP2), size = 1.2,
               arrow = arrow(length = unit(0.5, "cm"))) +
   geom_vline(xintercept = c(0), color = "grey70", linetype = 2) +
   geom_hline(yintercept = c(0), color = "grey70", linetype = 2) +
  geom_text(data = ccavectors, aes(x = CAP1*1.5, y = CAP2*1.5, 
                                   label = rownames(ccavectors)),
                                   #nudge_x = 0.3, nudge_y = 0.3
                                   size=6 ) +
  theme_bw() +
  scale_fill_manual(values= pal3) +
  scale_color_manual(values=pal3) +
  labs(x = "CAP1 [1.6%]", y="CAP2 [0.7%]") +
  stat_ellipse(aes(color=cat), size = 1, alpha = 0.5) + 
  #stat_ellipse(aes(fill = cat), geom="polygon", level=0.95, alpha=0.09) +
  labs( color = "Temperature \ncategory", fill = "Temperature category", 
        title = "Bray distances of hellinger transformed samples (n = 191), 180 classes.") +
  
     theme(legend.background = element_blank(),
           legend.box.background = element_rect(colour = "gray")) 

plot_cca


```

# PART IV: DS
```{r}
#| eval: false


# the DS dataset from EVA
Eva_master <- read_excel("C:/Users/Angela Cukusic/Desktop/Downloads/2022_Masterjoints.xlsx")
Eva_master$sample_season <- paste(Eva_master$Probennr, 
                                  "spring" , sep = "_")

# just spring DS data
main_sheet <- read.csv("C:/Users/Angela Cukusic/Desktop/Master thesis/data/main_sheet.csv")
main_sheet$sample_season <- paste(main_sheet$Sample.ID, 
                                  main_sheet$Season, sep = "_")
data_DS <- left_join(Eva_master, main_sheet, by = "sample_season") %>% 
            dplyr::filter(grepl('DS', Geol_beb)) 
data_DS <- ens2 %>% rownames_to_column("JMF.ID") %>% right_join(., data_DS, by ="JMF.ID")

################################################

add_data3 <- read_excel("C:/Users/Angela Cukusic/Desktop/Downloads/2022_01_31_Daten_Andela(1).xlsx")
data_DS <- add_data1[, c("Sample_ID", "Name_HBC")] %>% 
  full_join(., add_data3,  by = "Name_HBC") %>% 
  dplyr::select(!"Name_HBC") %>% left_join(data_DS , . , 
                                           by = c("Sample.ID" = "Sample_ID"),
                                           relationship = "many-to-many") 
data_DS["heatsources_UHI"] <- lapply(data_DS["heatsources_UHI"], as.factor) 
data_DS["heatsources_GWWP"] <- lapply(data_DS["heatsources_GWWP"], as.factor) 
data_DS["heatsources_FW"] <- lapply(data_DS["heatsources_FW"], as.factor) 
data_DS["heatsources_Versie"] <- lapply(data_DS["heatsources_Versie"], as.factor) 
data_DS["heatsources_count"] <- lapply(data_DS["heatsources_count"], as.factor) 
data_DS["OF_beb"] <- lapply(data_DS["OF_beb"], as.factor) 

#add it to new_metadata
new_metadata <- add_data1[, c("Sample_ID", "Name_HBC")] %>% 
  full_join(., add_data3,  by = "Name_HBC") %>% 
  dplyr::select(!"Name_HBC") %>% left_join(new_metadata , . , 
                                           by = c("Sample.ID.x" = "Sample_ID"),
                                           relationship = "many-to-many") 
new_metadata["heatsources_UHI"] <- lapply(new_metadata["heatsources_UHI"], as.factor) 
new_metadata["heatsources_GWWP"] <- lapply(new_metadata["heatsources_GWWP"], as.factor) 
new_metadata["heatsources_FW"] <- lapply(new_metadata["heatsources_FW"], as.factor) 
new_metadata["heatsources_Versie"] <- lapply(new_metadata["heatsources_Versie"], as.factor) 
new_metadata["heatsources_count"] <- lapply(new_metadata["heatsources_count"], as.factor) 



#just for spring


#combining eva master to shannon needs the HBT name to JMF
#of_beb

# add HBT_name to the Sample ID
new_metadata <- add_data1[, c("Sample_ID", "Name_HBC")] %>% 
  left_join(new_metadata, .,  by = c("Sample.ID.x" = "Sample_ID"))

full_data1  <-   ens2 %>% 
  rownames_to_column("JMF.ID") %>% 
  left_join(new_metadata,. , by = "JMF.ID") %>% 
  dplyr::select("Name_HBC"  , "Shannon")
  

Eva_master <- read_excel("C:/Users/Angela Cukusic/Desktop/Downloads/2022_Masterjoints.xlsx")



data <- full_data1 %>% 
  right_join(., Eva_master,  by = "Name_HBC") %>% 
 dplyr::filter(grepl('DS', Geol_beb)) 
data$OF_beb[data$OF_beb == 3] <- 0 #changing the 3s to 0
data["OF_beb"] <- lapply(data["OF_beb"], as.factor) 

split_labels <- str_wrap(c("no effect", "running water effect", "stagnant water effect"), width = 8)

p1 <- data %>% dplyr::filter(OF_beb != "NA") %>% 
  ggplot(aes(x=OF_beb, y=Shannon)) + geom_boxplot(fill = "gray", alpha=0.5) +
  theme_bw() + labs(x="Surface water impact", y= "Shannons diversity index") +
  scale_x_discrete(labels = split_labels)



shapiro.test (data_for_alpha2[which(data_for_alpha2$GWWP == "1"),"ens"] ) # norm
shapiro.test ( data_for_alpha2[which(data_for_alpha2$GWWP == "0"),"ens"]) # norm
bartlett.test(ens ~ GWWP, data= data_for_alpha2) # homo
summary( aov(ens ~ GWWP, data= data_for_alpha2) )




#wastewater
data["wastew_beb"] <- lapply(data["wastew_beb"], as.factor) 

  p2 <- data %>% 
 #dplyr::filter(OF_beb != "NA") %>% 
  ggplot(aes(x=wastew_beb, y=Shannon)) + geom_boxplot(fill = "gray", alpha=0.5) +
 theme_bw() + labs(x="Waste water impact", y= "Shannons diversity index") +
 scale_x_discrete(labels = c("no effect", "effect present", "unclear"))

p1 + p2

#redox
data["Redox_beb"] <- lapply(data["Redox_beb"], as.factor) 

split_labels <- str_wrap(c("o2 > 2", "o2 < 2", "Mn>0.15 & Fe>0.15", "s>0.1"), width = 8)

p3 <- data %>% 
 dplyr::filter(Redox_beb != "NA") %>% 
  ggplot(aes(x=Redox_beb, y=Shannon)) + geom_boxplot(fill = "gray", alpha=0.5) +
  theme_bw() + labs(x="Redox cascade", y= "Shannons diversity index") +
 scale_x_discrete(labels = split_labels )
                     


p1 + p2 +p3



##################################################################

data2 <- left_join(data, new_metadata[which(new_metadata$Season == "spring"),c("JMF.ID", "Name_HBC")], by = "Name_HBC")
unique_values  <-  unique(data2$JMF.ID)
data3 <- data2[!duplicated(data2$JMF.ID,  fromLast = TRUE), ] 
data3 <- data3[which(data3$JMF.ID != "NA"), ] 

data3[order(data3$JMF.ID), ]

rownames(data3) <- data3$JMF.ID
asv_ds_spring <- srs_asv_all[, which(colnames(srs_asv_all) %in% rownames(data3))]


#categorise by temperature groups
data3$Cat3 <- cut(data3$T_cali, 
                         breaks=c(0, 10, 12, 14, 16, 18, 20, 30), 
                         labels=c('<10', '10-12', '12-14', '14-16', 
                                  '16-18', '18-20', '20<'))
# find reference samples (the one in Marchfeld)
reference <- new_metadata %>% 
  filter(sample_season %in% c("22-246_fall", "EM10_spring",  "EM62_spring", "22-52_fall", 
                              "22-52_spring", "22.7/2_spring", "22.7/2_fall"))

# distinguish reference samples from the 10-12C
data3 <- data3 %>%
  mutate( reference = case_when(JMF.ID %in% reference$JMF.ID ~ "referential", 
                                TRUE ~ "other")) 
data3$ref_cat <- paste(data3$Cat3, data3$reference, sep="_")
data3 <- data3 %>% 
  mutate( category = case_when(ref_cat == "<10_other" ~ "<10",
                               ref_cat == "10-12_other" ~ "10-12",
                               ref_cat == "12-14_other" ~ "12-14",
                               ref_cat == "14-16_other" ~ "14-16",
                               ref_cat == "16-18_other" ~ "16-18",
                               ref_cat == "18-20_other" ~ "18-20",
                               ref_cat == "20<_other" ~ "20<",
                               TRUE ~ "ref_cat")) %>% 
  dplyr::select(!c("Cat3" , "reference" ,    "ref_cat"))
#at this point 281 samples







ps_all <- phyloseq(otu_table(as.matrix(asv_ds_spring), taxa_are_rows=TRUE), 
                   tax_table(as.matrix(new_taxonomy) ) , 
                  sample_data(as.data.frame(data3)) )

ps_all <- prune_taxa(taxa_sums(ps_all) > 0, ps_all)

```


## ordination
```{r}
#| eval: false

# perform ordination
unconstrained_aitchison_pca <- ps_all %>% tax_fix() %>% 
  tax_filter(min_prevalence = 0.1, tax_level = "Genus") %>%
  tax_agg("Family") %>%
  tax_transform("clr") %>%
  ord_calc()
 #ord_calc will automatically infer you want a "PCA" here
 #specify explicitly with method = "PCA", or you can pick another method

# create plot
pca_plot <- unconstrained_aitchison_pca %>%
 ord_plot(
    plot_taxa = 1:6, colour = "bmi_group", size = 1.5,
   tax_vec_length = 0.325,
  tax_lab_style = tax_lab_style(max_angle = 90, aspect_ratio = 0.5),
 auto_caption = 8)

# customise plot
customised_plot <- pca_plot +
  stat_ellipse(aes(linetype = bmi_group, colour = bmi_group), linewidth = 0.3) + # linewidth not size, since ggplot 3.4.0
  scale_colour_brewer(palette = "Set1") +
theme(legend.position = "bottom") +
coord_fixed(ratio = 0.5, clip = "off") # makes rotated labels align correctly

# show plot
customised_plot




```


```{r}
#| eval: false

data_for_reg <- cbind(new_metadata) %>% dplyr::select(where(is.numeric)) %>% 
 scale(center = T, scale = T) %>% cbind(., richness, ens) %>% data.frame() %>% 
dplyr::select(! c("DIC", "SO4", "Cl", "Na", "Ca", "Mg", "Lf", "richness",
                 "NO3", "NH3", "Fe", "X", "Y", "DOM_bix") )


nullModel = lm(ens ~ 1, data=data_for_reg) # model with the intercept only
fullModel = lm(ens ~ ., data=data_for_reg) # model with all  variables

#choose
step.model<-stepAIC(fullModel, # start with a model containing no variables
                   direction = 'both', # run forward selection
                  trace = 0) # dont show step-by-step process of model selection
summary(step.model) # only temp significant
RsquareAdj(step.model) # 0.04038098 -> 4 %
summary(lm(ens ~ Temp, data = data_for_reg)) 


### other models ####


# categorical variables
new_metadata %>% dplyr::select(!where(is.numeric)) 

# turn NAs from OF_beb into 0
new_metadata["OF_beb"] <- lapply(new_metadata["OF_beb"], as.character)
# Replace NA with 0
new_metadata[is.na(new_metadata)] <- 0 
# Change character columns back to factors
new_metadata["OF_beb"] <- lapply(new_metadata["OF_beb"], as.factor) 

# another thing needs fixing:
# add data about heat sources i forgot before
add_data3 <- read_excel("~/Desktop/DS_analysis/data/heat_sources.xlsx")
new_metadata <- add_data1[, c("Sample_ID", "Name_HBC")] %>% 
 full_join(., add_data3,  by = "Name_HBC") %>% 
dplyr::select(!"Name_HBC") %>% left_join(new_metadata, . , 
                                        by = c("Sample_ID.x" = "Sample_ID") ) 
new_metadata["heatsources_UHI"] <- lapply(new_metadata["heatsources_UHI"], as.factor) 
new_metadata["heatsources_GWWP"] <- lapply(new_metadata["heatsources_GWWP"], as.factor) 
new_metadata["heatsources_FW"] <- lapply(new_metadata["heatsources_FW"], as.factor) 
new_metadata["heatsources_Versie"] <- lapply(new_metadata["heatsources_Versie"], as.factor) 
new_metadata["heatsources_count"] <- lapply(new_metadata["heatsources_count"], as.factor) 

boxplot(richness ~ new_metadata$Geol_beb)
boxplot(richness ~ new_metadata$heatsources_UHI) # according to how many heat sources impact well
summary(aov(richness ~ category*Land.use*season*Geol_beb*UHI*heatsources_UHI, data=new_metadata) ) 
# category:Geol_beb:heatsources_UHI interaction significant
#summary(aov(richness ~ category:Geol_beb:heatsources_UHI, data=new_metadata) )
#fit3 <- lm(richness ~ category:Geol_beb:heatsources_UHI, data=new_metadata) 
#summary(fit3 ) # not sign


# plot interactions
#cat_plot(fit3, pred = Geol_beb, modx = heatsources_UHI, geom = "line")



# the same but for ens
#boxplot(ens ~ new_metadata$Geol_beb)
#boxplot(ens ~ new_metadata$UHI)
#summary(aov(ens ~ category*Land.use*season*Geol_beb*UHI, data=new_metadata) ) 
# category + UHI + Land.use:UHI significant
#model_ens <- aov(ens ~ category + UHI + Land.use:UHI, data=new_metadata) 
#summary(model_ens)
# i should do a kruskal wallis?






```

